app/dashboard/views.py

```diff
@@ -5362,41 +5378,57 @@ def validate_number(user_id, twilio, phone, redis):
             'msg': 'Phone type isn\'t supported'
         }, status=401)
 
-    verification = twilio.verify.verifications.create(to=phone, channel='sms')
-    redis.set(f'verification:{user_id}:timestamp', timezone.now().isoformat())
-    redis.set(f'verification:{user_id}:phone', phone)
+    twilio.verify.verifications.create(to=phone, channel='sms')
+    profile.last_validation_request = timezone.now()
+    profile.validation_attempts += 1
+    profile.encoded_number = hash_number
+    profile.save()
+
+    ProfileVerification.objects.create(profile=user.profile,
+                                       caller_type=validation.caller_name[
+                                           "caller_type"] if validation.caller_name else '',
+                                       carrier_error_code=validation.carrier['error_code'],
+                                       mobile_network_code=validation.carrier['mobile_network_code'],
+                                       mobile_country_code=validation.carrier['mobile_country_code'],
+                                       carrier_name=validation.carrier['name'],
+                                       carrier_type=validation.carrier['type'],
+                                       country_code=validation.country_code,
+                                       phone_number=hash_number)
+
+    redis.set(f'verification:{user.id}:phone', phone)
```

Comment:
I dislike store the number in the DB since we need the number in plain format to validate the code the user submits, so storing the number Redis will act as temp storage.